controller.py

```diff
@@ -54,4 +56,88 @@ def run(quad, des_state):
                    k_p_theta * (des_theta - theta) + k_d_theta * (q_des - q),
                    k_p_psi * (des_psi - psi) + k_d_psi * (r_des - r)]]).T
 
+    print(F)
+    print(M)
     return F, M
+
+def run_LQR(quad, des_state):
+
+    # get drone state
+    x, y, z = quad.position()
+    x_dot, y_dot, z_dot = quad.velocity()
+    phi, theta, psi = quad.attitude()
+    p, q, r = quad.omega()
+
+    # get desired state
+    des_x, des_y, des_z = des_state.pos
+    des_x_dot, des_y_dot, des_z_dot = des_state.vel
+    des_x_ddot, des_y_ddot, des_z_ddot = des_state.acc
+    des_psi = des_state.yaw
+    des_psi_dot = des_state.yawdot    
+
+
+    # For the LQR controller, the input to the controller are the states -x-,
+    # and reference values -r-, and the output -u- (inputs to the drone) 
+    # ar the Thrust and desired moments, (F,M) that should be applied to 
+    # the drone
+    #
+    # In general    u = Nu*r - K(x -Nx*r) = -K*x + (Nu + K*Nx)*r
+    # where K are the LQR gains, Nu and Nx are reference input matrices.
+    # See more at p.493 "Feedback Control of Dynamic Systems, Franklink G,
```

Comment:
Good summery, could you write some simple description on how you tune and design lqr matrix, and their effect on the control in general?